test/Kestrel.Core.Tests/HttpParserTests.cs

```diff
@@ -386,6 +386,59 @@ public void ParseRequestLineSplitBufferWithoutNewLineDoesNotUpdateConsumed()
             Assert.Equal(buffer.End, examined);
         }
 
+        [Fact]
+        public void ParseHeadersWithGratuitouslySplitBuffers()
+        {
+            var parser = CreateParser(_nullTrace);
+            var buffer = ReadOnlySequenceFactory.CreateSegments(
```

Comment:
Use https://github.com/aspnet/Common/blob/dev/shared/Microsoft.Extensions.Buffers.Testing.Sources/ReadOnlySequenceFactory.cs#L16 ?

Reply:
I dunno how to use your crazy libraries 😄